docs/docs/api/turn-by-turn/api-reference.md

```diff
@@ -276,6 +288,7 @@ Directions options should be specified at the top level of the JSON object.
 | `language` | The language of the narration instructions based on the [IETF BCP 47](https://tools.ietf.org/html/bcp47) language tag string. If no language is specified or the specified language is unsupported, United States-based English (en-US) is used. [Currently supported language list](#supported-language-tags) |
 | `directions_type` |  An enum with 3 values. <ul><li>`none` indicating no maneuvers or instructions should be returned.</li><li>`maneuvers` indicating that only maneuvers be returned.</li><li>`instructions` indicating that maneuvers with instructions should be returned (this is the default if not specified).</li></ul> |
 | `format` | Four options are available: <ul><li>`json` is default valhalla routing directions JSON format</li><li>`gpx` returns the route as a GPX (GPS exchange format) XML track</li><li>`osrm` creates a OSRM compatible route directions JSON</li><li>`pbf` formats the result using protocol buffers</li></ul> |
+| `shape_format` | If the format is `osrm`: Specifies the optional format for the path shape of each connection. One of `polyline6` (default), `polyline5`, `geojson` or `no_shape`. |
```

Comment:
shape_format can’t have the value „osrm“ IIRC. Also this is only implemented for some endpoints, I think trace_attributes and matrix only, so it’d rather belong there. Or you checked and it’s even implemented for /route?

Reply:
I've just checked it again. Its little messy :D
- Matrix has it (except TDMatrix?!) and its already documented.. 
- Turn by turn also has it if you specify `"format":"osrm"`. I've just gave it a test and code can be seen here: https://github.com/valhalla/valhalla/blob/afd01613cd430bdc9550cfd7352dea7552004be5/src/tyr/route_serializer_osrm.cc#L280C1-L280C12
- Elevation has it and its documented.
- For trace_attributes I couldn't find anything. But I also didn't try it out.

Reply:
Ah right elevation I meant, not trace_attributes.. Ok cool, didn’t know about the option for osrm, but I guess it makes sense as its default is polyline5. Then never mind!